avalanche/training/supervised/strategy_wrappers.py

```diff
@@ -276,6 +281,187 @@ def __init__(
         )
 
 
+class GenerativeReplay(SupervisedTemplate):
+    """Generative Replay Strategy
+
+    This implements Deep Generative Replay for a Scholar consisting of a Solver
+    and Generator as described in https://arxiv.org/abs/1705.08690.
+
+    The model parameter should contain the solver. As an optional input
+    a generator can be wrapped in a trainable strategy 
+    and passed to the generator_strategy parameter. By default a simple VAE will
+    be used as generator.
+
+    For the case where the Generator is the model itself that is to be trained,
+    please simply add the GenerativeReplayPlugin() when instantiating 
+    your Generator's strategy.
+
+    See GenerativeReplayPlugin for more details.
+    This strategy does not use task identities.
+    """
+
+    def __init__(
+        self,
+        model: Module,
+        optimizer: Optimizer,
+        criterion=CrossEntropyLoss(),
+        train_mb_size: int = 1,
+        train_epochs: int = 1,
+        eval_mb_size: int = None,
+        device=None,
+        plugins: Optional[List[SupervisedPlugin]] = None,
+        evaluator: EvaluationPlugin = default_evaluator,
+        eval_every=-1,
+        generator_strategy: BaseTemplate = None,
+        **base_kwargs
+    ):
```

Comment:
Really like the idea of having 2 different strategies, one for the classifier (this one) and one for the generative model (passed as an argument). This allows a great generalization.